src/moveCues.ts

```diff
@@ -0,0 +1,93 @@
+import { Rundown, RundownCueOrderItem } from '@rundown-studio/types'
+
+/**
+ * Moves selected cues to a new position within the rundown's cue order.
+ *
+ * @async
+ * @function
+ * @param {RundownCueOrderItem[]} cueOrder - The original array of cue order items.
+ * @param {Array<Rundown['id']>} selectedCues - IDs of the cues to move.
+ * @param {string} destination - Target position formatted as "mainIndex.subIndex".
+ * @throws {Error} If a group header is attempted to be moved inside another group.
+ * @return {Promise<RundownCueOrderItem[]>} - The reordered cue array.
+ */
+export async function moveCues (
+  cueOrder: RundownCueOrderItem[],
+  selectedCues: Array<Rundown['id']>,
+  destination: string,
+): Promise<RundownCueOrderItem[]> {
+  const selectedCueIds = new Set(selectedCues)
+  const selectedItems = selectCueItems(cueOrder, selectedCueIds)
+  const [mainIndex, subIndex] = destination.split('.').map(Number)
+
+  // Prevent moving a group header into another group
+  if (selectedItems.some((item) => item.children) && subIndex !== undefined) {
+    throw new Error('Cannot move a group header inside another group')
+  }
+
+  const newCueOrder: RundownCueOrderItem[] = []
+  let currentIndex = 0
+
+  if (mainIndex === 0) newCueOrder.push(...selectedItems)
+
+  for (const item of cueOrder) {
+    currentIndex++
+
+    if (!item.children) {
+      if (!selectedCueIds.has(item.id)) newCueOrder.push(item)
+      if (currentIndex === mainIndex && subIndex === undefined) {
+        newCueOrder.push(...selectedItems)
+      }
+    } else {
+      const children = []
+      let childIndex = 0
+
+      if (currentIndex === mainIndex && subIndex === 0) {
+        children.push(...selectedItems)
+      }
+
+      for (const child of item.children) {
+        childIndex++
+        if (!selectedCueIds.has(child.id)) children.push(child)
+        if (currentIndex === mainIndex && childIndex === subIndex) {
+          children.push(...selectedItems)
+        }
+      }
+
+      if (mainIndex === currentIndex && subIndex > childIndex) {
+        children.push(...selectedItems)
+      }
+
+      if (!selectedCueIds.has(item.id)) {
+        newCueOrder.push({ id: item.id, children })
+      }
+
+      if (currentIndex === mainIndex && subIndex === undefined) {
+        newCueOrder.push(...selectedItems)
+      }
+    }
```

Comment:
1. Add some comments for the outer if/else condition to make clear what cases you are handling there.
2. The loop has several instances of `newCueOrder.push(...selectedItems)` that, on first sight, are NOT in mutually exclusive conditions. (aka. if -> if else -> else). That makes the function non-intuitive.

Improve readability:
- Bail early, use `continue` to jump to the next loop as early as possible.
- Use negative if-conditions, meaning, use them to bail early (with `continue`) instead of positive conditions. This avoids repeated `newCueOrder.push(...selectedItems)`.
- Use if-else to make it clearer what case excludes the other.

You can feed these instructions to an AI.

Reply:
AI not smart enough... ⬇️
<img width="436" alt="image" src="https://github.com/user-attachments/assets/50265d26-7dba-45fd-9584-cf8bb29bfb94">

Anyway, jumping to next loop or avoid repeated `newCueOrder.push(...selectedItems)` seems like a good idea but in practice in each iteration we need to get to the end, because the same loop adds both the selected items as adds the ones that stay in position again. So it has to go through the end.

Basically in each iteration not only one thing is made, but at least two and in some cases even three:
1. Keep cues that are not selected for moving.
2. If correct position, add cues selected to be moved.
3. If group, maybe insert before group (move to 3.0), maybe insert at the end (move to 3.∞).

And the pairs 1-2 or 1.3 are not exclusive, so or I separate in two different loops, or I can't early continue.